drivers/wifi/nrf700x/CMakeLists.txt

```diff
@@ -162,38 +162,56 @@ else()
 	message(FATAL_ERROR "QSPI node not found in dts")
 endif()
 
-
-dt_nodelabel(nrf70_fw_node NODELABEL "nrf70_fw_partition")
-if (NOT DEFINED nrf70_fw_node)
-	message(FATAL_ERROR "nrf70_fw_partition node not found in dts")
-endif()
-dt_prop(nrf70_fw_offset PATH ${nrf70_fw_node} PROPERTY reg INDEX 0)
-if (NOT DEFINED nrf70_fw_offset)
-	message(FATAL_ERROR "nrf70_fw_partition.reg not found in dts")
-endif()
-math(EXPR NRF70_PATCH_ADDR "${xip_addr} + ${nrf70_fw_offset}")
-
-find_package(Python3)
-# Using python module instead of intelhex command line tool to handle windows
-execute_process(
-	COMMAND ${Python3_EXECUTABLE}
-		-c "import sys; import intelhex; intelhex.bin2hex(sys.argv[1], sys.argv[2], int(sys.argv[3]))"
-		${NRF70_PATCH}
+if (CONFIG_PARTITION_MANAGER_ENABLED)
+	add_custom_target(nrf70_wifi_fw_patch_target
+		DEPENDS ${CMAKE_BINARY_DIR}/zephyr/nrf70.hex
+	)
+	add_custom_command(OUTPUT ${CMAKE_BINARY_DIR}/zephyr/nrf70.hex
+		COMMAND bin2hex.py  --offset $<TARGET_PROPERTY:partition_manager,nrf70_wifi_fw_XIP_ABS_ADDR> ${NRF70_PATCH} ${CMAKE_BINARY_DIR}/zephyr/nrf70.hex
```

Comment:
you cannot just assume bin2hex is just in the user path

Reply:
Right, I have used the `python3_executable` earlier, but faced some issue here, let me check.